example/all-custom/index.js

```diff
@@ -0,0 +1,49 @@
+import ActionMap from "../../src/action/ActionMap.js";
+import ActionManager from "../../src/action/ActionManager.js";
+
+import AddValueFilter from "./AddValueFilter.js";
+import DummyInputSource from "./DummyInputSource.js";
+
+/**
+ * This demonstrates how to set up a completely custom ActionManager, using none of the default input sources, maps, or filters.
+ */
+export default function initInput() {
+  // Create an action manager without defaults
+  let actionManager = new ActionManager(false);
+
+  actionManager.addInputSource(
+    "dummy", // will be mapped to /input/dummy/0/
+    new DummyInputSource([
+      { path: "moo", parameters: { goo: "doo" } }, // Will generate /input/dummy/0/moo inputs
+      { path: "nums", parameters: { value: 22 } } // Will generate /input/dummy/0/nums inputs
+    ])
+  );
+
+  // Add a custom filter that adds value based on filter parameters in the JSON binding
+  actionManager.addFilter("add-value", new AddValueFilter());
+
+  // Add and switch to a custom action map
+  actionManager.addActionMap("custom", new ActionMap([...actionManager.filters], "./custom.json"));
```

Comment:
The idea is that the ActionManager starts out empty and the app writer decides which action maps to add for different situations like different display modes (so, when in flat mode you look for keyboard but in immersive mode you look for wand buttons) and in different experience modes (like when you're playing the game you map different inputs than if you're looking at the configuration menus).

Reply:
I guess that we could pass the filters into ActionMap.update(...) each time but I'm not sure why that's better.

Reply:
Ok I thought that once you added a filter to your app it will get registered directly on the actionmanager (for example similar to when you include an aframe component, that it's automatically being registered).
but that's fine I just wanted to clarify it

Reply:
Yeah, actually the more I think about it the more I think it's weird to have the filters registered in the ActionManager and then again in the ActionMaps. Maybe we should just have the ActionManager pass the filters into ActionMap.update(...).
My original thought was that each ActionMap might have filters that are configured in different ways, but it ended up that the filterParameters take care of most of that concern.